### DISCUSSION TITLE
How to ignore property on Insert/Update/Migrate but not on read.

### DISCUSSION THREAD

**Discussion by youssefbennour** [ServiceStack/Discuss#81] [Q&A]
I have a model which I want to correspond to a temporal table in the database, since it's not possible to generate temporal tables by default in OrmLite, I have added some custom SQL to support temporal tables.
Now my problem is that the temporal table has 2 extra date time columns : CreatedAt, ChangedAt.
I want to have those two columns as properties in my model but when inserting / updating / migrating, I want it to be ignored, however I want it to able to read it.

**Comment by mythz** [comment]
You can use the `[IgnoreOnInsert]` and `[IgnoreOnUpdate]` attributes.

**Reply by youssefbennour** [reply]
@mythz 
But does that prevent the migration from generating those properties as columns in the table ?

**Reply by mythz** [reply]
What's the actual issue that you're having?

**Reply by youssefbennour** [reply]
So after I've learned about the [IgnoreOnInsert] and [IgnoreOnUpdate], the remaining problem, is that I want to ignore some properties from a model, before running the migration.
Why would I want to do this, because after running the migration, I want to make the generated tables temporal (i.e. I will add 2 date time columns manually : 'CreatedAt' and 'ChangedAt' , which will be managed automaitcally by the database ), At the same time I want to be able to read those 2 extra columns along the model.

**Reply by mythz** [reply]
The data models embedded in your [code-first migrations](https://docs.servicestack.net/ormlite/db-migrations) is not the same model that's used in your App, so just use the correct data models for the behavior you want.
